File: app/admin/router/system/activateCode.go
package system

import (
	"github.com/gin-gonic/gin"
	"haocean/health-enforcement/app/admin/api/system"
	"haocean/health-enforcement/app/core/utils/jwt"
)

// InitActivateCode 初始化激活码管理路由
func InitActivateCode(e *gin.Engine) {
	v := e.Group("system")
	{
		auth := v.Group("")
		auth.Use(jwt.JWTAuthMiddleware())
		{
			auth.GET("/activate-code/list", system.ListActivateCode)
			auth.GET("/activate-code/:id", system.GetActivateCode)
			auth.POST("/activate-code/generate", system.GenerateActivateCode)
			auth.DELETE("/activate-code/:ids", system.DeleteActivateCode)
			auth.PUT("/activate-code/disable/:id", system.DisableActivateCode)
		}
	}
}

File: frontend/health-enforcement-ui/src/api/system/activateCode.js
import request from '@/utils/request'
import { parseStrEmpty } from "@/utils/ruoyi";

// 查询激活码列表
export function listActivateCode(query) {
  return request({
    url: '/system/activate-code/list',
    method: 'get',
    params: query
  })
}

// 查询激活码详细
export function getActivateCode(codeId) {
  return request({
    url: '/system/activate-code/' + parseStrEmpty(codeId),
    method: 'get'
  })
}

// 生成激活码
export function generateActivateCode(data) {
  return request({
    url: '/system/activate-code/generate',
    method: 'post',
    data: data
  })
}

// 删除激活码
export function deleteActivateCode(codeIds) {
  return request({
    url: '/system/activate-code/' + codeIds,
    method: 'delete'
  })
}

// 禁用激活码
export function disableActivateCode(codeId) {
  return request({
    url: '/system/activate-code/disable/' + codeId,
    method: 'put'
  })
}
